spec/brokers_spec.cr

```diff
@@ -36,6 +37,34 @@ module PlaceOS::Api
           updated.name.should_not eq original_name
         end
       end
+
+      describe "scopes" do
+        test_scope(Model::Broker, base, "brokers")
+
+        it "checks scope on update" do
+          _, authorization_header = authentication(scope: [PlaceOS::Model::UserJWT::Scope.new("brokers", PlaceOS::Model::UserJWT::Scope::Access::Write)])
+          broker = Model::Generator.broker.save!
+          original_name = broker.name
+          broker.name = UUID.random.to_s
+
+          id = broker.id.as(String)
+          path = base + id
+
+          result = update_route(path, broker, authorization_header)
+
+          result.status_code.should eq 200
+          updated = Model::Broker.from_trusted_json(result.body)
+
+          updated.id.should eq broker.id
+          updated.name.should_not eq original_name
+
+          _, authorization_header = authentication(scope: [PlaceOS::Model::UserJWT::Scope.new("brokers", PlaceOS::Model::UserJWT::Scope::Access::Read)])
+          result = update_route(path, broker, authorization_header)
+
+          result.success?.should be_false
+          result.status_code.should eq 403
+        end
```

Comment:
Is this not part of `test_scope`?

Reply:
If not, since the pattern is repeated a bit, maybe it can be pulled into a `test_update_write_scope` macro or something less verbose